From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Martijn Muijsers <martijnmuijsers@live.nl>
Date: Wed, 12 Jul 2023 21:29:08 +0200
Subject: [PATCH] Dev import deobfuscation fixes

Contains all fixes to source code files after decompilation (that are then to be modified by Fiddle but not Paper), needed to make them function as intended again after re-compilation.

License: GPL-3.0 (https://www.gnu.org/licenses/gpl-3.0.html)
Fiddle - https://fiddlemc.org

diff --git a/src/main/java/net/minecraft/network/protocol/game/ClientboundUpdateRecipesPacket.java b/src/main/java/net/minecraft/network/protocol/game/ClientboundUpdateRecipesPacket.java
index ff6323bd5170359cf9c0d004aba68a792fa6a382..ddb04a1e7d50024ee2f3692bccf471509720f65e 100644
--- a/src/main/java/net/minecraft/network/protocol/game/ClientboundUpdateRecipesPacket.java
+++ b/src/main/java/net/minecraft/network/protocol/game/ClientboundUpdateRecipesPacket.java
@@ -44,9 +44,9 @@ public class ClientboundUpdateRecipesPacket implements Packet<ClientGamePacketLi
         return new RecipeHolder<>(resourceLocation2, recipe);
     }
 
-    public static <T extends Recipe<?>> void toNetwork(FriendlyByteBuf buf, RecipeHolder<?> recipe) {
+    public static <T extends Recipe<?>> void toNetwork(FriendlyByteBuf buf, RecipeHolder<T> recipe) { // Fiddle - dev import deobfuscation fixes
         buf.writeResourceLocation(BuiltInRegistries.RECIPE_SERIALIZER.getKey(recipe.value().getSerializer()));
         buf.writeResourceLocation(recipe.id());
-        recipe.value().getSerializer().toNetwork(buf, recipe.value());
+        ((net.minecraft.world.item.crafting.RecipeSerializer<T>) recipe.value().getSerializer()).toNetwork(buf, recipe.value()); // Fiddle - dev import deobfuscation fixes
     }
 }
diff --git a/src/main/java/net/minecraft/world/level/block/Blocks.java b/src/main/java/net/minecraft/world/level/block/Blocks.java
index 9a2de546dc2af2ad4bf5d32ca6583f0e1f3f70d8..4d5bc4d46fa90de1d89e15bd5dc053d22e07a3d9 100644
--- a/src/main/java/net/minecraft/world/level/block/Blocks.java
+++ b/src/main/java/net/minecraft/world/level/block/Blocks.java
@@ -1215,7 +1215,7 @@ public class Blocks {
     }
 
     private static Boolean ocelotOrParrot(BlockState state, BlockGetter world, BlockPos pos, EntityType<?> type) {
-        return (boolean)type == EntityType.OCELOT || type == EntityType.PARROT;
+        return type == EntityType.OCELOT || type == EntityType.PARROT; // Fiddle - dev import deobfuscation fixes
     }
 
     private static Block bed(DyeColor color) {
